src/MCStack.cpp

```diff
@@ -228,15 +306,55 @@ MCStack::getFirstEnabledTransition()
     // their state members on
     const bool transitionIsInSleepSet = sTop.threadIsInSleepSet(i);
     if (transitionIsEnabled && !transitionIsInSleepSet)
+      // FIXME:  We are returning an address from this local call frame.
+      //         This is dangerous, since '&nextTransitionForI' is a ptr.
+      //         Did we intend that the return type here should be
+      //         'const MCTransition &' ?
+      //         See discussion "This is more C++ obfuscation.", above.
       return &nextTransitionForI;
   }
   return nullptr;
 }
 
+// =====================================================================
+/*****************************************************************
+ * Below here, we are computing the enabled sets to be stored in *
+ * backgrack sets in the history (the stack).                 *
+ *****************************************************************/
 std::unordered_set<tid_t>
 MCStack::getCurrentlyEnabledThreads()
 {
   std::unordered_set<tid_t> enabledThreadsInState;
+  // FIXME:  We are returning an unordered set.
+  //         An address to it might be more performant.
+
+  // FIXME:  This code for ENV_PRINT_AT_TRACE_SEQ exists twice because
+  //         we already had duplicate "paragraph".  We should merge this
+  //         into common code in a future, single utility function,
+  //         and we return traceSeq[traceIdx++] from the "else" branch
+  //         of this code, to be used in the following paragraph.
+  static bool print_at_trace_seq = (getenv(ENV_PRINT_AT_TRACE_SEQ) != NULL);
+  if (print_at_trace_seq) {
+    static int traceIdx = 1; // traceSeq[0] is for thread 0 'starts'. Skip it.
+    if (traceIdx <= 1) {
+      trace_string_to_int_array(getenv(ENV_PRINT_AT_TRACE_SEQ), traceSeq,
+                                sizeof(traceSeq) / sizeof(traceSeq[0]));
+    }
+    if (traceSeq[traceIdx] == -1) {
+      mcprintf("*** END OF trace sequence\n");
+      mcprintf("*** ... continuing beyond trace sequence\n");
+      print_at_trace_seq = false;
+      assert(traceId == 0);
+      setenv(ENV_PRINT_AT_TRACE_ID, "0", 1); // but his was also set earlier
```

Comment:
This prevents McMini from properly continuing DPOR after the trace executes. In particular, you're telling McMini to halt execution at trace 0 after following the first `N` steps as you have added in this PR. To let McMini to continue its DPOR search for bugs without backtracking on the prefix, you should simply leave this environment variable untouched.